easybuild/easyblocks/x/xmipp.py

```diff
@@ -0,0 +1,152 @@
+            '--config=force',
+            'profile=no',
+            'fast=yes',
+            'warn=no',
+            'release=yes',
+            'gtest=no',
+            'cuda=no',
+            'debug=no',
+            'matlab=no',
+            'java=no',
+            'LINKERFORPROGRAMS="$CXX"',
+            'MPI_BINDIR="{mpi_bindir}"',
+            'JAVA_HOME="{java_home}"',
+            'JAVAC=javac',
+            'CC="$CC"',
+            'CXXFLAGS="$CXXFLAGS -DMPICH_IGNORE_CXX_SEEK -I$EBROOTPYTHON/include/python{short_python_ver}" \
+                      "-I{python_libdir}/numpy/core/include/"',
+            'CXX="$CXX"',
+            'MPI_CC="$MPICC"',
+            'MPI_CXX="$MPICXX"',
+            'MPI_INCLUDE="$MPI_INC_DIR"',
+            'MPI_LIBDIR="$MPI_LIB_DIR"',
+            'MPI_LINKERFORPROGRAMS="$MPICC"',
+            'LIBPATH="$LD_LIBRARY_PATH"',
+            '{configopts}',
+        ])
+        cmd.format(
+            parallel=self.cfg['parallel'],
+            short_python_ver=self.python_short_ver,
+            python_libdir=python_libdir,
+            preconfigopts=self.cfg['preconfigopts'],
+            configopts=self.cfg['configopts'],
+            mpi_bindir=os.path.join(get_software_root(self.toolchain.MPI_MODULE_NAME[0]), 'bin'),
+            java_home=java_root
+        )
+        run_cmd(cmd, log_all=True, simple=True)
+
+    def build_step(self):
+        """Custom build procedure for Xmipp: call the scons wrapper with compile argument"""
+        cmd = ' '.join([
+            'LD_LIBRARY_PATH=$LD_LIBRARY_PATH:$PWD/lib',
+            'python external/scons/scons.py',
+            'mode=compile -j {parallel}',
+        ])
+        cmd.format(parallel=self.cfg['parallel'])
+        run_cmd(cmd, log_all=True, simple=True)
+
+    def install_step(self):
+        """install step for xmipp, this builds a local database and seems to do some tests?"""
+        cmd = ' '.join([
+            'XMIPP_HOME=$PWD',
+            'PATH=$PWD/bin:$PATH',
+            'PYTHONPATH="$PYTHONPATH:$PWD/protocols:$PWD/libraries/bindings/python/:' \
+            '$EBROOTPYTHON/lib/python{short_python_ver}/lib-dynload/"',
+            'python setup.py install'
+        ])
+        cmd.format(short_python_ver=self.python_short_ver)
+        run_cmd(cmd, log_all=True, simple=True)
+
+    def sanity_check_step(self):
+        """Custom sanity check for Xmipp."""
+        custom_paths = {
+            'files': ['xmipp_%s' % x for x in ['imagej', 'mpi_run', 'phantom_create', 'tomo_project', 'volume_align']],
+            'dirs': ['lib'],
+        }
+        super(EB_Xmipp, self).sanity_check_step(custom_paths=custom_paths)
+
+    def make_module_extra(self):
+        """Define Xmipp specific variables in generated module file, i.e. XMIPP_HOME."""
+        txt = super(EB_Xmipp, self).make_module_extra()
+        txt += self.module_generator.set_environment('XMIPP_HOME', '$root')
```

Comment:
I don't think we have a generic alternative for this yet (as in, something that will work for both Tcl and Lua module files), @pforai may correct me here

so stick with this `$root` approach, for now